ctapipe/io/hessiofilereader.py

```diff
@@ -0,0 +1,211 @@
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.core import Provenance
+from ctapipe.io.eventfilereader import EventFileReader
+from ctapipe.io.containers import DataContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription
+
+
+class HessioFileReader(EventFileReader):
+    """
+    EventFileReader for the hessio data format.
+
+    This class utilises `pyhessio` to read the hessio file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config, tool, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            from pyhessio import open_hessio
+            from pyhessio import close_file
+            from pyhessio import HessioGeneralError
+        except ImportError:
+            msg = "The `pyhessio` python module is required to access MC data"
+            self.log.error(msg)
+            raise
+
+        self.open_hessio = open_hessio
+        self.close_hessio = close_file
+        self.HessioGeneralError = HessioGeneralError
+
+        self.allowed_tels = None
+
+        if HessioFileReader._count > 0:
+            self.log.warn("Only one pyhessio reader allowed at a time. "
+                          "Previous hessio file will be closed.")
+            self.close_hessio()
+        HessioFileReader._count += 1
+
+        self._metadata['is_simulation'] = True
+
+    @staticmethod
+    def is_compatible(file_path):
+        return file_path.endswith('.gz')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close_hessio()
+
+    def _generator(self):
+        with self.open_hessio(self.input_path) as pyhessio:
+            # the container is initialized once, and data is replaced within
+            # it after each yield
+            Provenance().add_input_file(self.input_path, role='dl0.sub.evt')
+            counter = 0
+            eventstream = pyhessio.move_to_next_event()
+            if self.allowed_tels is not None:
+                self.allowed_tels = set(self.allowed_tels)
+            data = DataContainer()
+            data.meta['origin'] = "hessio"
+
+            # some hessio_event_source specific parameters
+            data.meta['input'] = self.input_path
```

Comment:
I think `data.meta["input_path"]` is more clear.

Reply:
could be a url in case of zmq input

Reply:
So maybe use fully qualified urls `file:path` `http://<>/``` etc

Reply:
year in that case this should maybe be called `input_url` .. I just though `input` is not very specific.

Reply:
Actually, I think *path* is confusing, since we use the word to mean "a list of directories to search, separated by colons" (like `$PATH`) elsewhere in the code, and it doesn't mean that here.  Why not, `input_url`, since it can be more than just a filename? (URLs work in general, at least for pyhessio, and we could easily make the others work).

Reply:
that comment actually applies to all other uses of `input_path` where it doesn't refer to a search path. For the Container metadata, it doesn't really matter, since we now store that in the Provenance system.

Reply:
While I fully agree that `input_url` should be used, simply because also urls can be specified here. I think using "path" for "a list of directories to be searched in" is confusing:

I have never seen this definition for "path". In Python it seams to be common to use the plural for a list, or any general container of stuff, hence I would recommend using `paths` in such a case. If once wants to convey, that theses paths should be used for searching something, maybe `search_paths` or `path_candidates`, when this group contains many paths, but only one will ultimately be correct.

Also: None of these have a similar definition of `path`.

* http://searchnetworking.techtarget.com/definition/path
* https://techterms.com/definition/path
* https://en.wikipedia.org/wiki/Path_(computing)